content/blogposts/2022/sourcegraph-backstage-bootstrapping-catalog-adoption-batch-changes.md

```diff
@@ -0,0 +1,189 @@
+
+```diff
+# app-config.yaml
+  catalog:
+	locations:
+-  	- type: url
+-    	  target: https://github.com/my-org/my-repo/blob/main/components.yaml
+```
+
+### Using Sourcegraph to open pull requests
+
+Backstage Software catalog is designed to empower developers to own the metadata of the components they use. It provides a GitOps-friendly way of pulling component metadata from repositories into the catalog's database. Information like the component's name, owner, type, and other metadata can be described via the `catalog-info.yaml` file. Developers can edit this file in their repository and Backstage will automatically read the changed file to update the metadata in the database.
+
+To avoid the chicken-and-egg problem while setting up Backstage, we can bootstrap the creation of `catalog-info.yaml` files with Sourcegraph's [`Batch Changes`](https://docs.sourcegraph.com/batch_changes). Sourcegraph will automatically open pull requests against every repository to add a `catalog-info.yaml` file.
+
+The benefit of this approach is that repository owners can either just merge the pull request without additional work. Or, even if they want to change the description, add links, or update the owners listed in the `catalog-info.yaml` file before merging the pull request, at least the pull request was started. 
+
+Moreover, Sourcegraph Batch Changes let you track all PRs to completion. So, if you want to open PRs against all repos but don't know the ownership information, you can start the process, then use the dashboard to figure out who you need to follow up with to make sure everything gets merged. 
+
+Let's see what this batch change would look like. Sourcegraph's Batch Changes are described by a spec that consist of three main parts:
+
+* A search query
+* Steps to execute to generate the diff
+* A pull request template
+
+Here’s an example of a batch change spec for adding `catalog-info.yaml` files to your repositories:
+
+```yml
+name: generate-catalog-infos-for-backstage
+description: Batch change to add repositories to the Backstage catalog
+
+# All repos from a given codehost
+on:
+  - repositoriesMatchingQuery: repo:github\.com
+
+steps:
+  # Run comby over the search results in each repository:
+  - run: |
+  	apk add github-cli;
+  	gh auth login -h github.com;
+  	gh repo view sourcegraph/sourcegraph --json=description,url,name,owner,nameWithOwner,languages > tmp.json;
+  	python /tmp/create-catalog.py;
+  	rm tmp.json
+	container: python:3-alpine
+	files:
+  	/tmp/create-catalog.py: |
+    	import json
+    	TEMPLATE ="""kind: Component
+    	metadata:
+      	name: {}
+      	description: |
+        	{}
+      	links:
+      	- title: Repository
+        	url: {}
+      	annotations:
+        	github.com/project-slug: {}
+    	"""
```

Comment:
Something looks invalid about this YAML. There is a ton of mixed tabs and spaces, which may be the reason. YAML requires spaces, tabs don't work. This would be a hard-to-catch problem when these snippets are copy-pasted by readers.